diff --git a/ios/DomWebView.swift b/ios/DomWebView.swift
index 27ed0444fca82b170b21a9d5f7562f2fb4885178..e7cc671fa364c7a10be2783df619422ff1520516 100644
--- a/ios/DomWebView.swift
+++ b/ios/DomWebView.swift
@@ -69,9 +69,10 @@ internal final class DomWebView: ExpoView, UIScrollViewDelegate, WKUIDelegate, W
     }
 
     if let source,
-      let request = RCTConvert.nsurlRequest(source.toDictionary(appContext: appContext)),
-      webView.url?.absoluteURL != request.url {
-      webView.load(request)
+      let uri = source.uri,
+      let url = URL(string: uri),
+      webView.url?.absoluteURL != url {
+      webView.load(URLRequest(url: url))
     }
   }
 
